deploy/test/test_cases/TEST_ID_27_push_to_file.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -euxo pipefail
+
+deploy_push_to_file() {
+  configure_conjur_url
+
+  echo "Running Deployment push to file"
+
+  if [[ "$DEV" = "true" ]]; then
+    ./dev/config/k8s/secrets-provider-init-push-to-file.sh.yml > ./dev/config/k8s/secrets-provider-init-push-to-file.yml
+    $cli_with_timeout apply -f ./dev/config/k8s/secrets-provider-init-push-to-file.yml
+
+    $cli_with_timeout "get pods --namespace=$APP_NAMESPACE_NAME --selector app=init-env --no-headers | wc -l"
+  else
+    wait_for_it 600 "$CONFIG_DIR/test-env-push-to-file.sh.yml | $cli_without_timeout apply -f -" || true
+
+    expected_num_replicas=`$CONFIG_DIR/test-env-push-to-file.sh.yml |  awk '/replicas:/ {print $2}' `
+
+    # Deployment (Deployment for k8s and DeploymentConfig for Openshift) might fail on error flows, even before creating the pods. If so, re-deploy.
+    if [[ "$PLATFORM" = "kubernetes" ]]; then
+        $cli_with_timeout "get deployment test-env -o jsonpath={.status.replicas} | grep '^${expected_num_replicas}$'|| $cli_without_timeout rollout latest deployment test-env"
+    elif [[ "$PLATFORM" = "openshift" ]]; then
+        $cli_with_timeout "get dc/test-env -o jsonpath={.status.replicas} | grep '^${expected_num_replicas}$'|| $cli_without_timeout rollout latest dc/test-env"
+    fi
+
+    echo "Expecting for $expected_num_replicas deployed pods"
+    $cli_with_timeout "get pods --namespace=$APP_NAMESPACE_NAME --selector app=test-env --no-headers | wc -l | grep $expected_num_replicas"
+  fi
+}
+echo "Deploying Push to file tests"
+
+create_secret_access_role
+create_secret_access_role_binding
+
+echo "Deploying test_env without CONTAINER_MODE environment variable"
+export CONTAINER_MODE_KEY_VALUE=$KEY_VALUE_NOT_EXIST
+deploy_push_to_file
+
+echo "Expecting secrets provider to succeed as an init container"
+pod_name="$(get_pod_name "$APP_NAMESPACE_NAME" 'app=test-env')"
+
+$cli_with_timeout "get pod $pod_name --namespace=$APP_NAMESPACE_NAME | grep -c 1/1"
+kubectl get deploy --namespace=$APP_NAMESPACE_NAME
+kubectl get pods --namespace=$APP_NAMESPACE_NAME
+
+$cli_with_timeout "logs $pod_name -c cyberark-secrets-provider-for-k8s"
+
+declare -a file_data=("\"url\": \"postgresql://test-app-backend.app-test.svc.cluster.local:5432\"
```

Comment:
Using Bash associative arrays (`declare -A ...`), and using single quotes for defining file content (so that you don't have to escape the double quotes in the content) here would be helpful, I think. Here's an example script:

```
#!/bin/bash

FILES="foo1 foo2 foo3"

declare -A expected_content
expected_content[foo1]='This is file "foo1", line 1
This is file "foo1", line 2'
expected_content[foo2]='This is file "foo2", line 1
This is file "foo2", line 2'
expected_content[foo3]='This is file "foo3", line 1
This is file "foo3", line 2'

declare -A file_format
file_format[foo1]="yaml"
file_format[foo2]="json"
file_format[foo3]="template"

i=0
test_failed=false
for f in $FILES; do
    format="${file_format[$f]}"
    echo "Checking file $f content, file format: $format"
    content="$(cat $f)"
    if [ "$content" == "${expected_content[$f]}" ]; then
        echo "Push to File PASSED for file format $format!"
    else
        echo "Push to File FAILED for file format $format!"
        echo "Expected content:"
        echo "================="
        echo "${expected_content[$f]}"
        echo 
        echo "Actual content:"
        echo "==============="
        echo "$content"
        echo
        test_failed=true
    fi
done

if "$test_failed"; then
    exit 1
fi
```